# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: security/sandbox/chromium/sandbox/win/src/named_pipe_policy.cc
# Commit: 370c40e49230
# Full Hash: 370c40e4923073a87845b57c1f4851bacd9663da
# Author: Bob Owen <bobowencode@gmail.com>
# Date: 2019-06-12 21:42:35
# Regressor Bug: 1552160
# File Overlap Count: 2
# Description:
#   Bug 1552160 Part 1: Roll-up of chromium sandbox update and mozilla patches to get a running browser. r=jld,aklotz,tjr,bobowen
#   
#   This updates security/sandbox/chromium/ files to chromium commit 84108231f6e6e0772fb9a4643679ce76aa771e67.
#   
#   Existing and new patches applied from security/sandbox/chromium-shim/patches/with_update/ to give a compiling and mostly working browser.
# ==============================================================================

diff -r bd5ecd72f76c -r 370c40e49230 security/sandbox/chromium/sandbox/win/src/named_pipe_policy.cc
--- a/security/sandbox/chromium/sandbox/win/src/named_pipe_policy.cc	Wed Jun 12 21:55:59 2019 +1200
+++ b/security/sandbox/chromium/sandbox/win/src/named_pipe_policy.cc	Wed Jun 12 11:10:48 2019 +0100
@@ -15,22 +15,24 @@
 
 // Creates a named pipe and duplicates the handle to 'target_process'. The
 // remaining parameters are the same as CreateNamedPipeW().
-HANDLE CreateNamedPipeHelper(HANDLE target_process, LPCWSTR pipe_name,
-                              DWORD open_mode, DWORD pipe_mode,
-                              DWORD max_instances, DWORD out_buffer_size,
-                              DWORD in_buffer_size, DWORD default_timeout,
-                              LPSECURITY_ATTRIBUTES security_attributes) {
-  HANDLE pipe = ::CreateNamedPipeW(pipe_name, open_mode, pipe_mode,
-                                   max_instances, out_buffer_size,
-                                   in_buffer_size, default_timeout,
-                                   security_attributes);
+HANDLE CreateNamedPipeHelper(HANDLE target_process,
+                             LPCWSTR pipe_name,
+                             DWORD open_mode,
+                             DWORD pipe_mode,
+                             DWORD max_instances,
+                             DWORD out_buffer_size,
+                             DWORD in_buffer_size,
+                             DWORD default_timeout,
+                             LPSECURITY_ATTRIBUTES security_attributes) {
+  HANDLE pipe = ::CreateNamedPipeW(
+      pipe_name, open_mode, pipe_mode, max_instances, out_buffer_size,
+      in_buffer_size, default_timeout, security_attributes);
   if (INVALID_HANDLE_VALUE == pipe)
     return pipe;
 
   HANDLE new_pipe;
-  if (!::DuplicateHandle(::GetCurrentProcess(), pipe,
-                         target_process, &new_pipe,
-                         0, FALSE,
+  if (!::DuplicateHandle(::GetCurrentProcess(), pipe, target_process, &new_pipe,
+                         0, false,
                          DUPLICATE_CLOSE_SOURCE | DUPLICATE_SAME_ACCESS)) {
     return INVALID_HANDLE_VALUE;
   }
@@ -60,8 +62,9 @@
 
 DWORD NamedPipePolicy::CreateNamedPipeAction(EvalResult eval_result,
                                              const ClientInfo& client_info,
-                                             const base::string16 &name,
-                                             DWORD open_mode, DWORD pipe_mode,
+                                             const base::string16& name,
+                                             DWORD open_mode,
+                                             DWORD pipe_mode,
                                              DWORD max_instances,
                                              DWORD out_buffer_size,
                                              DWORD in_buffer_size,
@@ -73,10 +76,9 @@
     return ERROR_ACCESS_DENIED;
   }
 
-  *pipe = CreateNamedPipeHelper(client_info.process, name.c_str(),
-                                open_mode, pipe_mode, max_instances,
-                                out_buffer_size, in_buffer_size,
-                                default_timeout, NULL);
+  *pipe = CreateNamedPipeHelper(client_info.process, name.c_str(), open_mode,
+                                pipe_mode, max_instances, out_buffer_size,
+                                in_buffer_size, default_timeout, nullptr);
 
   if (INVALID_HANDLE_VALUE == *pipe)
     return ERROR_ACCESS_DENIED;